scripts/extrinsic_check.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+set -e
+
+HEAD_BIN=./target/release/creditcoin-node
+HEAD_WS=ws://localhost:9944
+RELEASE_WS=ws://localhost:9945
+
+runtimes=(
+  "creditcoin-node"
+)
+
+# First we fetch the latest released binary
+latest_release() {
+  curl -s "https://api.github.com/repos/$1/releases/latest" | jq -r '.tag_name'
+}
+
+latest_release=$(latest_release 'gluwa/creditcoin')
+RELEASE_BIN="./creditcoin-node"
+echo "[+] Fetching binary for Creditcoin version $latest_release"
+curl -L "https://github.com/gluwa/creditcoin/releases/download/$latest_release/creditcoin-v${latest_release}-x86_64-unknown-linux-gnu.tar.gz"  --output creditcoin.tar.gz && tar -xzf creditcoin.tar.gz || exit 1
+chmod +x "$RELEASE_BIN"
+git fetch --depth="${GIT_DEPTH:-100}" origin 'refs/tags/*:refs/tags/*'
+
+
+for RUNTIME in "${runtimes[@]}"; do
+  echo "[+] Checking runtime: ${RUNTIME}"
+
+  release_transaction_version=$(
+    git show "tags/$latest_release:runtime/src/lib.rs" | \
+      grep 'transaction_version'
+  )
+
+  current_transaction_version=$(
+    grep 'transaction_version' "./runtime/src/lib.rs"
+  )
+
+  echo "[+] Release: ${release_transaction_version}"
+  echo "[+] Ours: ${current_transaction_version}"
+
+  if [ ! "$release_transaction_version" = "$current_transaction_version" ]; then
+    echo "[+] Transaction version for ${RUNTIME} has been bumped since last release."
+    exit 0
+  fi
+
+  # Start running the nodes in the background
+  $HEAD_BIN --chain=./chainspecs/qaSpec.json --tmp &
+  $RELEASE_BIN --chain=./chainspecs/qaSpec.json --ws-port 9945 --tmp &
+  jobs
+
+  # Sleep a little to allow the nodes to spin up and start listening
+  TIMEOUT=30
+  for i in $(seq $TIMEOUT); do
+    sleep 1
+      if [ "$(lsof -nP -iTCP -sTCP:LISTEN | grep -c '994[45]')" == 2 ]; then
+        echo "[+] Both nodes listening"
+        break
+      fi
+      if [ "$i" == $TIMEOUT ]; then
+        echo "[!] Both nodes not listening after $i seconds. Exiting"
+        jobs -p | xargs kill; sleep 5
+        exit 1
+      fi
+  done
```

Comment:
Isn't this what `./integration-tests/wait-for-creditcoin.sh` is supposed to do (minus the yarn install) ? 

Better to reuse that script and pass the URL as argument.

Reply:
It checks for ports that are opened.

The script compiles and builds a creditcoin-node, then it downloads the latest release from github. It then starts the two nodes at different ports.

Reply:
But still, it waits for the 2 nodes in a similar fashion like `wait-for-creditcoin.sh` does. Is my understanding correct ?

Reply:
Can you start making use of the `wait-for-creditcoin.sh` script instead of duplicating the same steps? The script accepts a target URL as an argument.